mono_repo/lib/src/commands/github/github_yaml.dart

```diff
@@ -9,27 +9,80 @@ import 'package:pub_semver/pub_semver.dart';
 import '../../ci_shared.dart';
 import '../../package_config.dart';
 import '../../root_config.dart';
+import '../../user_exception.dart';
 import '../../yaml.dart';
 import '../ci_script/generate.dart';
-import '../shared.dart';
+import 'self_validate_job.dart';
 
-String generateGitHubYml(
+Map<String, String> generateGitHubYml(
   RootConfig rootConfig,
   Map<String, String> commandsToKeys,
 ) {
-  final jobs = rootConfig.expand((config) => config.jobs);
+  final jobs = rootConfig.expand((config) => config.jobs).toList();
 
-  final jobList = Map.fromEntries([
-    if (rootConfig.monoConfig.selfValidateStage != null)
-      _selfValidateTaskConfig(),
-    ..._listJobs(jobs, commandsToKeys, rootConfig.monoConfig.mergeStages),
-  ]);
+  final selfValidateStage = rootConfig.monoConfig.selfValidateStage;
+  if (selfValidateStage != null) {
+    jobs.add(SelfValidateJob(selfValidateStage));
+  }
+
+  final allJobStages = jobs.map((e) => e.stageName).toSet();
+
+  final output = <String, String>{};
+
+  void doTheWork(
+    String fileName,
+    String workflowName,
+    Iterable<CIJob> myJobs,
+    MapEntry<String, Map<String, dynamic>> extraEntry,
+  ) {
+    if (output.containsKey(fileName)) {
+      throw UnimplementedError('Need a better error here!');
+    }
+    _jobNameCache.clear();
+    final jobList = Map.fromEntries([
+      if (extraEntry != null) extraEntry,
+      ..._listJobs(myJobs, commandsToKeys, rootConfig.monoConfig.mergeStages),
+    ]);
 
-  return '''
-${createdWith()}${toYaml(rootConfig.monoConfig.github.generate())}
+    output[fileName] = '''
+${createdWith()}${toYaml(rootConfig.monoConfig.github.generate(workflowName))}
 
 ${toYaml({'jobs': jobList})}
 ''';
+  }
+
+  final workflows = rootConfig.monoConfig.github.workflows;
+
+  if (workflows != null) {
+    for (var entry in workflows.entries) {
+      final myJobs = jobs
+          .where((element) => entry.value.stages.contains(element.stageName))
+          .toList();
+
+      if (myJobs.isEmpty) {
```

Comment:
This should really be checking that every listed stage has at least one job, not the sum of all jobs from all stages

Reply:
I thought that's what I'm doing here. `myJobs` is the set of jobs for the stages defined in that workflow.

If there are no jobs for that stage, then (clearly) we've configured a stage w/ no jobs.

Reply:
If I am reading it correctly this is the list of jobs for _all_ the stages in the workflow.

So if _any_ stage is valid (ie: any jobs are selected) then it will not complain.

If I list two stages, but one is a typo, then we also want to complain.

It is possible I am reading the code wrong though?

Reply:
See `.where((element) => entry.value.stages.contains(element.stageName))`

myJobs is filtered from the current entry